src/MessageDispatch.KurrentDB.Tests/SubscriberTests.cs

```diff
@@ -0,0 +1,526 @@
+using PharmaxoScientific.MessageDispatch.KurrentDB;
+using Testcontainers.EventStoreDb;
+
+namespace MessageDispatch.KurrentDB.Tests;
+
+public class SubscriberTests
+{
+    private const string StreamName = "stream1";
+    private string _connectionString;
+    private KurrentDBClient _kurrentDbClient;
+    private AwaitableDispatcherSpy _dispatcher;
+    private KurrentDbSubscriber? _subscriber;
+
+    [SetUp]
+    public async Task Setup()
+    {
+        const int eventStoreHostPort = 1234;
+        const string eventStoreVersion = "24.10.5";
+
+        var eventStoreImageName = RuntimeInformation.OSArchitecture == Architecture.Arm64
+            ? $"ghcr.io/eventstore/eventstore:{eventStoreVersion}-alpha-arm64v8"
+            : $"eventstore/eventstore:{eventStoreVersion}-bookworm-slim";
+
+        var eventStoreContainer = BuildEventStoreContainer(eventStoreImageName, eventStoreHostPort);
+        await eventStoreContainer.StartAsync();
+
+        var mappedHostPort = eventStoreContainer.GetMappedPublicPort(eventStoreHostPort);
+        _connectionString = $"esdb://admin:changeit@localhost:{mappedHostPort}?tls=true&tlsVerifyCert=false";
+
+        _kurrentDbClient = new KurrentDBClient(KurrentDBClientSettings.Create(_connectionString));
+        _dispatcher = new AwaitableDispatcherSpy();
+    }
+
+    [TearDown]
+    public async Task TearDown()
+    {
+        await _kurrentDbClient.DisposeAsync();
+        _subscriber?.ShutDown();
+    }
+
+    [Test]
+    public async Task CreateLiveSubscription_GivenNoEventsInStreamWhenNewEventsAdded_DispatchesEventsAndBecomesLive()
+    {
+        _subscriber = KurrentDbSubscriber.CreateLiveSubscription(
+            _kurrentDbClient,
+            _dispatcher,
+            StreamName,
+            new NullLogger<KurrentDbSubscriber>());
+
+        _subscriber.Start();
+
+        var event1 = SimpleEvent.Create();
+        var event2 = SimpleEvent.Create();
+        var event3 = SimpleEvent.Create();
+
+        List<SimpleEvent> events = [event1, event2, event3];
+
+        await AppendEventsToStreamAsync(event1, event2, event3);
+        await _dispatcher.WaitForEventsToBeDispatched(event1, event2, event3);
+
+        var deserializedDispatchedEvents =
+            _dispatcher.DispatchedEvents.Select(DeserializeEventData<SimpleEvent>);
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(deserializedDispatchedEvents, Is.EqualTo(events));
+            Assert.That(_subscriber.IsLive);
+        });
+    }
+
+    [Test]
+    public async Task CreateLiveSubscription_GivenExistingEventsInStreamWhenNewEventsAdded_DispatchesNewEventsAndBecomesLive()
+    {
+        _subscriber = KurrentDbSubscriber.CreateLiveSubscription(
+            _kurrentDbClient,
+            _dispatcher,
+            StreamName,
+            new NullLogger<KurrentDbSubscriber>());
+
+        var oldEvent1 = SimpleEvent.Create();
+        var oldEvent2 = SimpleEvent.Create();
```

Comment:
I _think_ there is something funky here. This test looks identical to the one below in terms of setup and expection. I think the events passed to `WaitForEventsToBeDispatched` needs to include the old events too? `WaitForEventsToBeDispatched` would return too early if there were more dispatched events than expected due to `!=`